views/job-roles/detail.njk

```diff
@@ -405,6 +452,220 @@ function dismissAlert(alertId) {
   }
 }
 
+function viewApplicationDetails(app) {
+  // Parse the notes field to extract structured information
+  const parseNotesData = (notes) => {
+    if (!notes) return {};
+    
+    const lines = notes.split('\n');
+    const data = {};
+    
+    lines.forEach(line => {
+      if (line.includes('Name:')) {
+        const fullName = line.replace('Name:', '').trim();
+        const nameParts = fullName.split(' ');
+        data.firstName = nameParts[0] || '';
+        data.lastName = nameParts.slice(1).join(' ') || '';
+      } else if (line.includes('Current Job Title:')) {
+        data.jobTitle = line.replace('Current Job Title:', '').trim();
+      } else if (line.includes('Years of Experience:')) {
+        data.yearsOfExperience = line.replace('Years of Experience:', '').trim();
+      } else if (line.includes('LinkedIn:')) {
+        data.linkedinUrl = line.replace('LinkedIn:', '').trim();
+      } else if (line.includes('CV:')) {
+        data.cvUrl = line.replace('CV:', '').trim();
+      } else if (line.includes('Comments:')) {
+        data.comments = line.replace('Comments:', '').trim();
+      }
+    });
+    
+    return data;
+  };
+
+  const parsedData = parseNotesData(app.notes);
+  
+  // Create modal content with application data directly (no API call needed)
+  const modalHTML = `
+    <div id="applicationModal" class="fixed inset-0 bg-black bg-opacity-60 backdrop-blur-sm flex items-center justify-center p-4" style="z-index: 9999;">
+      <div class="bg-white rounded-lg shadow-2xl max-w-4xl w-full max-h-[90vh] overflow-y-auto relative border-2 border-gray-200" onclick="event.stopPropagation()" style="z-index: 10000; background-color: white !important;">
+        <div class="sticky top-0 bg-white border-b-2 border-kainos-primary px-6 py-4 flex justify-between items-center" style="background-color: white !important;">
+          <div>
+            <h3 class="text-2xl font-bold text-kainos-primary">Application Details - ${app.displayName || 'Unknown Applicant'}</h3>
+            <p class="text-sm text-gray-500 mt-1">Press ESC or click outside to close</p>
+          </div>
+          <button id="closeModalBtn" class="btn btn-sm btn-circle btn-ghost hover:bg-red-100 hover:text-red-600 transition-colors">
+            <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" viewBox="0 0 20 20" fill="currentColor">
+              <path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd" />
+            </svg>
```

Comment:
Let's refactor we're basically using js to build a string with html inside.
Instead we could just have a dialog created in nunjucks and the just control the visibility of it with js